### PR TITLE
fix: onOptionChange on TreeSelects and fix Switch Toggle

### LINKED ISSUE #9078: [Bug]: onOptionChange when user selects Download 2 files are downloaded at once in Multitreeselect & Treeselect widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Two files are getting downloaded when the user changes selection from drop down. This is observed in Multitreeselect & Treeselect widget

https://loom.com/share/17c90b9d98914ea2810cb350a33549e9

**Expected Behavior:** Only one file must get downloaded 

### Steps To Reproduce

1. Drag drop Multitree select widget
2. In property entity > Actions > onOptionChange > Download > Data to download: sample > File name with extension: ext > Type: Html(or any)
3. Go back to multitree select widget drop down > from drop down select a different option.

### Environment

Release

### Version

Cloud

### LINKED ISSUE #9598: [Bug]: By toggling the switch widget the toggle turns into blue colour
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

It is observed that when the user disables the switch widget toggle the turns into blue colour 

[LOOM DEMO](https://www.loom.com/share/8503a1e97ed640a78375739b6f75f047)

### Steps To Reproduce

1.Launch appsmith app in release/Prod Environment 
2.Drag and drop switch widget 
3. Disable the switch toggle(Mouse hover on the switch toggle so that the mouse icon will be a pointer and click on the toggle and observe the colour changes to blue)

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description
 FIxes onOptionChange for TreeSelect and Multiselect

Fixes #9078
Fixes #9598

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/OnOptionChange-called-twice-in-Tree-select 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/widgets/SwitchWidget/component/index.tsx | 93.75 **(0.42)** | 62.5 **(0)** | 50 **(0)** | 93.75 **(0.42)**</details>

### CODE REVIEW

File: app/client/src/widgets/SwitchWidget/component/index.tsx [hunk 1]
```diff
@@ -19,6 +20,10 @@ const SwitchComponentContainer = styled.div`
   align-items: center;
   .${Classes.CONTROL} {
     margin: 0;
+    input:checked ~ .${Classes.CONTROL_INDICATOR} {
```
Comment (sbalaji1192) [inline app/client/src/widgets/SwitchWidget/component/index.tsx:23]: Can we not achieve this without using important ?

Comment (Tooluloope) [reply app/client/src/widgets/SwitchWidget/component/index.tsx:23]: Fixed @sbalaji1192

**Review by sbalaji1192 — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@Tooluloope Tested this PR and both the issues #9078 and #9598 are working as expected

**Pull request merged**